src/main.dart

```diff
@@ -1,8 +1,138 @@
 // TODO: Run task list app
+
+import 'dart:io';
+import 'task.dart';
+
 void main(List<String> args) {
-  print('Hello World');
+  final TaskListApp taskApp = TaskListApp();
+
+  print('------------------ Task Manager ------------------');
+  print('------ Welcome to our beautfuil Task Manager -----');
+  print('------  Please enter the menu option below  ------');
+  print('--------------------------------------------------');
+
+  print('--- 1 for "List Task"');
+  print('--- 2 for "Add Task"');
+  print('--- 3 for "Edit Task"');
+  print('--- 4 for "Delete Task"');
+  print('--- -1 for "Exit App"');
+
+  while (true) {
+    print("");
+    print('Enter valid option: ');
+    int option = int.tryParse(stdin.readLineSync()!) ?? 0;
+
+    if (option == -1) break;
+
+    switch (option) {
+      case 1:
+        taskApp.listTasks();
+        break;
+      case 2:
+        taskApp.addTask();
+        break;
+      case 3:
+        taskApp.editTask();
+        break;
+      case 4:
+        taskApp.deleteTask();
+        break;
+      default:
+        print('Invalid option');
+    }
+  }
+
+  print("Thank you for using our app");
 }
 
 class TaskListApp {
```

Comment:
Since this is already a class, why not extract it to a different file? Code abstraction and refractoring is good.